bson/bsoncodec.go

```diff
@@ -276,28 +107,28 @@ func (fn ValueEncoderFunc) EncodeValue(ec EncodeContext, vw ValueWriter, val ref
 // ValueDecoder. A DecodeContext instance is provided and serves similar functionality to the
 // EncodeContext.
 type ValueDecoder interface {
-	DecodeValue(DecodeContext, ValueReader, reflect.Value) error
+	DecodeValue(DecoderRegistry, ValueReader, reflect.Value) error
```

Comment:
We don't want to remove `DecodeContext` from the `ValueDecoder` API, only from the `bson.Decoder` API. Keeping `DecodeContext` in the `DecodeValue` function may significantly simplify this PR.